generated/java/google-cloud-pubsub-v1/src/main/java/com/google/cloud/pubsub/spi/v1/SubscriberClient.java

```diff
@@ -1030,6 +1141,387 @@ public final void modifyPushConfig(ModifyPushConfigRequest request) {
+   *   // Do something
+   *   future.get();
+   * }
+   * </code></pre>
+   */
+  public final UnaryCallable<DeleteSnapshotRequest, Empty> deleteSnapshotCallable() {
+    return deleteSnapshotCallable;
+  }
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD
+  /**
+   * Seeks an existing subscription to a point in time or to a given snapshot, whichever is provided
+   * in the request.
+   *
+   * <p>Sample code:
+   *
+   * <pre><code>
+   * try (SubscriberClient subscriberClient = SubscriberClient.create()) {
+   *   String formattedSubscription = SubscriptionName.create("[PROJECT]", "[SUBSCRIPTION]").toString();
+   *   Timestamp time = Timestamp.newBuilder().build();
+   *   String snapshot = "";
+   *   SeekResponse response = subscriberClient.seek(formattedSubscription, time, snapshot);
+   * }
+   * </code></pre>
+   *
+   * @param subscription The subscription to affect.
+   * @param time The time to seek to. Messages retained in the subscription that were published
+   *     before this time are marked as acknowledged, and messages retained in the subscription that
+   *     were published after this time are marked as unacknowledged. Note that this operation
+   *     affects only those messages retained in the subscription (configured by the combination of
+   *     `message_retention_duration` and `retain_acked_messages`). For example, if `time`
+   *     corresponds to a point before the message retention window (or to a point before the
+   *     system's notion of the subscription creation time), only retained messages will be marked
+   *     as unacknowledged, and already-expunged messages will not be restored.
+   * @param snapshot The snapshot to seek to. The snapshot's topic must be the same as that of the
+   *     provided subscription. Format is `projects/{project}/snapshots/{snap}`.
+   * @throws com.google.api.gax.grpc.ApiException if the remote call fails
+   */
+  public final SeekResponse seek(String subscription, Timestamp time, String snapshot) {
+
+    SeekRequest request =
+        SeekRequest.newBuilder()
+            .setSubscription(subscription)
+            .setTime(time)
+            .setSnapshot(snapshot)
+            .build();
+    return seek(request);
+  }
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD
+  /**
+   * Seeks an existing subscription to a point in time or to a given snapshot, whichever is provided
+   * in the request.
+   *
+   * <p>Sample code:
+   *
+   * <pre><code>
+   * try (SubscriberClient subscriberClient = SubscriberClient.create()) {
+   *   String formattedSubscription = SubscriptionName.create("[PROJECT]", "[SUBSCRIPTION]").toString();
+   *   Timestamp time = Timestamp.newBuilder().build();
+   *   String snapshot = "";
+   *   SeekRequest request = SeekRequest.newBuilder()
+   *     .setSubscription(formattedSubscription)
+   *     .setTime(time)
+   *     .setSnapshot(snapshot)
+   *     .build();
```

Comment:
The sample code here is wrong - it is setting both `time` and `snapshot` in the one-of. /cc @michaelbausor